structuredheadings/plugin.js

```diff
@@ -730,31 +729,33 @@
 
           return [CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor)];
         },
-
-        exec: function (editor, value) {
-          var selection = editor.getSelection();
-          var headings = CKEDITOR.plugins.structuredheadings.getHeadingsInSelection(
-                editor,
-                selection
-              );
+        exec: function (editor, value) { // eslint-disable-line max-statements
+          var headingIterator = CKEDITOR.plugins.structuredheadings.getHeadingIteratorForSelection(
```

Comment:
(typically, an iterator exposes a getNextParagraph method.  In this case, the iterator itself is technically hidden inside a closer, and we epxose the getNextHeading method and badly name it the iterator)